clients/drcachesim/common/options.cpp

```diff
@@ -333,3 +333,39 @@ droption_t<bool> op_reuse_verify_skip
  "Verifies every skip list-calculated reuse distance with a full list walk. "
  "This incurs significant additional overhead.  This option is only available "
  "in debug builds.");
+
+#define OP_RECORD_FUNC_ITEM_SEP "&"
+// XXX i#3048: replace function return address with function callstack
+// XXX i#3048: add a section to drcachesim.dox.in on function tracing
+droption_t<std::string> op_record_function
+(DROPTION_SCOPE_ALL, "record_function", DROPTION_FLAG_ACCUMULATE,
+ OP_RECORD_FUNC_ITEM_SEP, "",
+ "Record invocations trace for the specified function(s).",
+ "Record invocations trace for the specified function(s) in the option"
+ " value. Default value is empty. The value should fit this format:"
+ " function_name|function_id|func_args_num"
+ " (example: -record_function \"memset|10|3\"). The trace would contain"
+ " information for function return address, function argument value(s),"
+ " and function return value. We only record pointer-sized arguments and"
+ " return value. The trace is labeled with the function_id via an ID entry"
+ " prior to each set of value entries."
+ " Note that the provided function id should be unique, and not collide with"
+ " existing heap functions (see -record_heap_value ) if record_heap"
+ " option is enabled.");
+droption_t<bool> op_record_heap
+(DROPTION_SCOPE_ALL, "record_heap", false,
+ "Enable recording trace for the defined heap functions.",
+ "It is a convenient option to enable recording trace for the defined heap"
+ " functions in -record_heap_value. Specify this option is equivalent to"
+ " -record_function [heap_functions], where [heap_functions] is"
+ " the value in -record_heap_value.");
+droption_t<std::string> op_record_heap_value
+(DROPTION_SCOPE_ALL, "record_heap_value", DROPTION_FLAG_ACCUMULATE,
+ OP_RECORD_FUNC_ITEM_SEP,
+ "malloc|0|1" OP_RECORD_FUNC_ITEM_SEP "free|1|1" OP_RECORD_FUNC_ITEM_SEP
+ "tc_malloc|2|1" OP_RECORD_FUNC_ITEM_SEP "tc_free|3|1" OP_RECORD_FUNC_ITEM_SEP
+ "__libc_malloc|4|1" OP_RECORD_FUNC_ITEM_SEP "__libc_free|5|1",
+ " The actual value of the defined heap functions.",
+ " The actual value of the defined heap functions. Value should fit the same"
+ " format required by -record_function. These functions will not"
```

Comment:
The separator is not described under -record_function, though, so that makes it a little confusing.  Probably best to explain the separator under -record_function.

Reply:
Done